src/pybind/rbd/rbd.pyx

```diff
@@ -1876,6 +1876,36 @@ class RBD(object):
                     if name]
         finally:
             free(c_names)
+    
+    def group_list(self, ioctx, name):
+        """
+        Get group's id.
+
+        :param ioctx: determines which RADOS pool is read
+        :type ioctx: :class:`rados.Ioctx`
+        :param name: which group
+        :type name: str
+        :returns: str -- group id
+        """
+        name = cstr(name, 'name')
+        cdef:
+            rados_ioctx_t _ioctx = convert_ioctx(ioctx)
+            char *_name = name
+            size_t size = 32
+            char *id = NULL
+        try:
+            while ret == -errno.ERANGE and size <= 4096:
+                id =  <char *>realloc_chk(id, size)
+                with nogil:
+                    ret = rbd_group_get_id(_ioctx, _name, id, &size)
+                if ret == -errno.ERANGE:
+                    size *= 2
```

Comment:
yes, The length of the group id will not exceed 32, so maybe there no need double the size and get id again

Reply:
I think it's better to be generic and still have the second call, even if we know that currently IDs are limited to 14 characters.  @ajarr's point is that instead of doubling, `size` could be used directly.

Reply:
I found second call is useful when I write test case, I no need care about size value, just use `size = 0`, `rbd_group_get_id` method will return right value.

Reply:
Right, but here, in the production code, starting with 32 instead of 0 is much better.  Since currently IDs are smaller than that, each Python API call would correspond to a single C API call.  If you start with 0, it would be two C API calls.